extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -421,28 +428,37 @@ export class AmpAnalytics extends AMP.BaseElement {
       return Promise.resolve();
     }
 
+    let requestPromise = Promise.resolve(request);
     // Add any given extraUrlParams as query string param
     if (this.config_['extraUrlParams'] || trigger['extraUrlParams']) {
       const params = Object.create(null);
       Object.assign(params, this.config_['extraUrlParams'],
           trigger['extraUrlParams']);
       for (const k in params) {
         if (typeof params[k] == 'string') {
-          params[k] = this.expandTemplate_(params[k], trigger, event);
+          requestPromise = requestPromise
+              .then(() => this.variableService_.expandTemplate(
+                  params[k], trigger, this.config_, event)
+              .then(value => { params[k] = value; }));
         }
       }
-      request = this.addParamsToUrl_(request, params);
+      requestPromise.then(() => {
+        request = this.addParamsToUrl_(request, params);
+      });
     }
 
-    this.config_['vars']['requestCount']++;
-    request = this.expandTemplate_(request, trigger, event);
-
-    // For consistency with amp-pixel we also expand any url replacements.
-    return urlReplacementsForDoc(this.win.document).expandAsync(request)
+    return requestPromise.then(() => {
+      this.config_['vars']['requestCount']++;
+      return this.variableService_.expandTemplate(
+          request, trigger, this.config_, event)
+        .then(request =>
+          // For consistency with amp-pixel we also expand any url replacements.
+          urlReplacementsForDoc(this.win.document).expandAsync(request))
```

Comment:
I'm getting some concerns over the refactoring that @mkhatib is currently doing. You've seen that, right?